### PR TITLE
Get patcher working on 1.16

### PR DESCRIPTION
**Pull request by coderbot16**
Things mostly seem to work, but obviously I can't actually test the patched mods yet. This was surprisingly easy, thanks to all of the work done ahead of time to prepare by Glitch.

For those watching: Please note that this doesn't actually let you use Patchwork as a whole on 1.16, since Patchwork API (a critical component) is still only available for 1.14.4. It will need to be ported to 1.16 as well first!

### CODE REVIEW

File: src/main/java/net/patchworkmc/patcher/util/ResourceDownloader.java [hunk 1]
```diff
@@ -113,6 +117,54 @@ public IMappingProvider setupAndLoadMappings(Path voldemapBridged) throws IOExce
...
+	 */
+	private void purgeNonexistentClassMappings(List<TsrgClass<RawMapping>> classes, Path mergedMinecraftJar) throws URISyntaxException, IOException {
+		// I'm using an iterator here since it allows us to remove entries easily as we iterate over the List
+		Iterator<TsrgClass<RawMapping>> classIterator = classes.iterator();
+
+		URI jarUri = new URI("jar:" + mergedMinecraftJar.toUri());
+
+		boolean needsClarificationMessage = false;
+
+		try (FileSystem fs = FileSystems.newFileSystem(jarUri, Collections.emptyMap())) {
```
Comment (ramidzkh) [inline src/main/java/net/patchworkmc/patcher/util/ResourceDownloader.java]: Isn't it possible to pass `mergedMinecraftJar` in place of the URI option?

Comment (coderbot16) [reply src/main/java/net/patchworkmc/patcher/util/ResourceDownloader.java]: Yep! Done.

**Review by ramidzkh — APPROVED** [review]
Looks good, LGTM

**Comment by coderbot16** [comment]
Thanks! I'm going to give Glitch a chance to look it over before merging it.

**Comment by coderbot16** [comment]
Glitch says that they don't have time to review PRs right now, so I'll just merge this.

**Pull request merged**